### PR TITLE
Convert controller sa files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 1]
```diff
@@ -295,27 +295,27 @@ func (e *TokensController) syncSecret() {
...
 		sa, saErr := e.getServiceAccount(secretInfo.namespace, secretInfo.saName, secretInfo.saUID, true)
 		switch {
 		case saErr != nil:
-			klog.Error(saErr)
+			klog.ErrorS(saErr, "Error")
 			retry = true
 		case sa == nil:
 			// Delete token
-			klog.V(4).Infof("syncSecret(%s/%s), service account does not exist, deleting token", secretInfo.namespace, secretInfo.name)
+			klog.V(4).InfoS("syncSecret, service account does not exist, deleting token", "secretNamespace", secretInfo.namespace, "secret", secretInfo.name)
```
Comment (yangjunmyfm192085) [inline pkg/controller/serviceaccount/tokens_controller.go:310]: Maybe you can use klog.KRef?

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 2]
```diff
@@ -377,7 +377,7 @@ func (e *TokensController) ensureReferencedToken(serviceAccount *v1.ServiceAccou
 	}
 	if liveServiceAccount.ResourceVersion != serviceAccount.ResourceVersion {
 		// Retry if our liveServiceAccount doesn't match our cache's resourceVersion (either the live lookup or our cache are stale)
-		klog.V(4).Infof("liveServiceAccount.ResourceVersion (%s) does not match cache (%s), retrying", liveServiceAccount.ResourceVersion, serviceAccount.ResourceVersion)
+		klog.V(4).InfoS("liveServiceAccount.ResourceVersion does not match cache, retrying", "serviceAccountResourceVersion", liveServiceAccount.ResourceVersion, "cachedServiceAccountResourceVersion", serviceAccount.ResourceVersion)
```
Comment (mikedanese) [inline pkg/controller/serviceaccount/tokens_controller.go:380]:
```suggestion
		klog.V(4).InfoS("liveServiceAccount.ResourceVersion does not match cache, retrying", "serviceAccountResourceVersion", liveServiceAccount.ResourceVersion, "resourceVersion", serviceAccount.ResourceVersion)
```

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 3]
```diff
@@ -295,27 +295,27 @@ func (e *TokensController) syncSecret() {
...
 			if retriable, err := e.deleteToken(secretInfo.namespace, secretInfo.name, secretInfo.uid); err != nil {
-				klog.Errorf("error deleting serviceaccount token %s/%s for service account %s: %v", secretInfo.namespace, secretInfo.name, secretInfo.saName, err)
+				klog.ErrorS(err, "Error deleting serviceaccount token for service account", "secretNamespace", secretInfo.namespace, "secret", secretInfo.name, "secretServiceAccountName", secretInfo.saName)
 				retry = retriable
 			}
 		default:
 			// Update token if needed
 			if retriable, err := e.generateTokenIfNeeded(sa, secret); err != nil {
-				klog.Errorf("error populating serviceaccount token %s/%s for service account %s: %v", secretInfo.namespace, secretInfo.name, secretInfo.saName, err)
+				klog.ErrorS(err, "Error populating service account tokens for service account", "secretNamespace", secretInfo.namespace, "secret", secretInfo.name, "secretServiceAccountName", secretInfo.saName)
```
Comment (mikedanese) [inline pkg/controller/serviceaccount/tokens_controller.go:318]:
```suggestion
				klog.ErrorS(err, "Error populating service account tokens for service account", "namespace", secretInfo.namespace, "secret", secretInfo.name, "name", secretInfo.saName)
```

Comment (mikedanese) [reply pkg/controller/serviceaccount/tokens_controller.go:318]: These logs look way more verbose. Can we try to keep the keys concise?

File: pkg/controller/serviceaccount/tokens_controller.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:310]: Fixed.

File: pkg/controller/serviceaccount/tokens_controller.go (hunk 3 above)
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:318]: Fixed with klog.KObj()

**Comment by scottilee** [comment]
@yangjunmyfm192085 and @mikedanese Thanks for you reviews! I've addressed your comments. Let me know if there's anything else I should fix.

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 4]
```diff
@@ -295,27 +295,27 @@ func (e *TokensController) syncSecret() {
...
+				klog.ErrorS(err, "Error")
 			}
 		}
 	default:
 		// Ensure service account exists
 		sa, saErr := e.getServiceAccount(secretInfo.namespace, secretInfo.saName, secretInfo.saUID, true)
 		switch {
 		case saErr != nil:
-			klog.Error(saErr)
+			klog.ErrorS(saErr, "Error")
```
Comment (aramase) [inline pkg/controller/serviceaccount/tokens_controller.go:306]:
```suggestion
			klog.ErrorS(saErr, "Error getting service account")
```

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 5]
```diff
@@ -238,28 +238,28 @@ func (e *TokensController) syncServiceAccount() {
 
 	saInfo, err := parseServiceAccountKey(key)
 	if err != nil {
-		klog.Error(err)
+		klog.ErrorS(err, "Error")
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:241]: Fix.ed

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 6]
```diff
@@ -279,14 +279,14 @@ func (e *TokensController) syncSecret() {
 
 	secretInfo, err := parseSecretQueueKey(key)
 	if err != nil {
-		klog.Error(err)
+		klog.ErrorS(err, "Error")
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:282]: Fix.ed

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 7]
```diff
@@ -295,27 +295,27 @@ func (e *TokensController) syncSecret() {
 			if err := clientretry.RetryOnConflict(RemoveTokenBackoff, func() error {
 				return e.removeSecretReference(secretInfo.namespace, secretInfo.saName, secretInfo.saUID, secretInfo.name)
 			}); err != nil {
-				klog.Error(err)
+				klog.ErrorS(err, "Error")
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:298]: Fixed.

File: pkg/controller/serviceaccount/tokens_controller.go (hunk 4 above)
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:306]: Fixed.

**Comment by scottilee** [comment]
@aramase Addressed your comments. Let me know if there is anything else!

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 8]
```diff
@@ -460,10 +460,10 @@ func (e *TokensController) ensureReferencedToken(serviceAccount *v1.ServiceAccou
 
 	if !addedReference {
 		// we weren't able to use the token, try to clean it up.
-		klog.V(2).Infof("deleting secret %s/%s because reference couldn't be added (%v)", secret.Namespace, secret.Name, err)
+		klog.V(2).InfoS("Deleting secret because reference couldn't be added", "secret", klog.KObj(secret), "err", err)
 		deleteOpts := metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &createdToken.UID}}
 		if err := e.client.CoreV1().Secrets(createdToken.Namespace).Delete(context.TODO(), createdToken.Name, deleteOpts); err != nil {
-			klog.Error(err) // if we fail, just log it
+			klog.ErrorS(err, "Error") // if we fail, just log it
```
Comment (navidshaikh) [inline pkg/controller/serviceaccount/tokens_controller.go:466]:
```suggestion
			klog.ErrorS(err, "Error deleting secrets") // if we fail, just log it
```

**Review by navidshaikh — CHANGES_REQUESTED** [review]
/wg structured-logging

File: pkg/controller/serviceaccount/tokens_controller.go (hunk 8 above)
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:466]: Fixed.

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 9]
```diff
@@ -238,28 +238,28 @@ func (e *TokensController) syncServiceAccount() {
 
 	saInfo, err := parseServiceAccountKey(key)
 	if err != nil {
-		klog.Error(err)
+		klog.ErrorS(err, "Error parsing service account key")
```
Comment (yuzhiquan) [inline pkg/controller/serviceaccount/tokens_controller.go:241]: msg maybe not needed.

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 10]
```diff
@@ -238,28 +238,28 @@ func (e *TokensController) syncServiceAccount() {
...
 		return
 	}
 
 	sa, err := e.getServiceAccount(saInfo.namespace, saInfo.name, saInfo.uid, false)
 	switch {
 	case err != nil:
-		klog.Error(err)
+		klog.ErrorS(err, "Error getting service account")
```
Comment (yuzhiquan) [inline pkg/controller/serviceaccount/tokens_controller.go:248]: ditto

File: pkg/controller/serviceaccount/tokens_controller.go (hunk 9 above)
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:241]: Already incorporated.

File: pkg/controller/serviceaccount/tokens_controller.go (hunk 10 above)
Comment (scottilee) [reply pkg/controller/serviceaccount/tokens_controller.go:248]: Already incorporated.

**Comment by scottilee** [comment]
Rebased.

**Comment by ritazh** [comment]
@mikedanese PTAL?

File: hack/.structured_logging [hunk 11]
```diff
@@ -1,2 +1,3 @@
 cmd/kubelet/...
+pkg/controller/serviceaccount/...
```
Comment (serathius) [inline hack/.structured_logging:2]: I would not recommend marking this directory as migrated we it makes collecting approvals harder and we didn't get approval from component owner

**Pull request closed without merging**